internal/ingress/controller/template/crossplane/http.go

```diff
@@ -55,6 +55,7 @@ func (c *crossplaneTemplate) initHTTPDirectives() ngx_crossplane.Directives {
 		buildDirective("map_hash_bucket_size", cfg.MapHashBucketSize),
 		buildDirective("proxy_headers_hash_max_size", cfg.ProxyHeadersHashMaxSize),
 		buildDirective("proxy_headers_hash_bucket_size", cfg.ProxyHeadersHashBucketSize),
+		buildDirective("proxy_pass_header", cfg.AllowBackendServerHeader),
```

Comment:
This part shouldn't be here. The right approach is the one you did below.

Looking at the template:

```
    {{ if $cfg.AllowBackendServerHeader }}
    proxy_pass_header Server;
    {{ end }}
```

the one you do below, checking and in case true appending is the right :) So just drop this line, and the other one is right! thanks!!

Reply:
Thanks, @rikatz! I will remove the first line.